llvm/tools/llvm-profgen/CSPreInliner.cpp

```diff
@@ -152,6 +152,9 @@ uint32_t CSPreInliner::getFuncSize(const ContextTrieNode *ContextNode) {
 }
 
 bool CSPreInliner::shouldInline(ProfiledInlineCandidate &Candidate) {
+  if (Candidate.SizeCost == 0)
+    return true;
+
```

Comment:
I saw `    SampleThreshold = SampleHotCallSiteThreshold * NormalizedHotness * 100 +
                      SampleColdCallSiteThreshold + 1;`, it should be > 0 number, so IIUC, the zero is from  `SampleColdCallSiteThreshold` which is set by default to zero. I wondering if we can just set it to 1, so we don't need an additional condition here. No strong option.

Reply:
+1. We probably don't want the special case. Either have +1 for cold path too, or remove +1 for hot path, and change to <= in the end.